src/test/regress/sql/alter_table_ao.sql

```diff
@@ -447,3 +447,38 @@ CREATE table ao_insert_empty_row (a integer, b text, c integer) WITH (APPENDONLY
 INSERT INTO ao_insert_empty_row SELECT 1,'a',1 FROM gp_id WHERE dbid=-999;
 -- assert non-empty seg entries
 SELECT * FROM gp_toolkit.__gp_aoseg('ao_insert_empty_row'::regclass);
+
+-- test case: alter AO table add partition, the preference of the storage setting is: the encoding clause > table setting > gp_default_storage_options
+SET gp_add_partition_inherits_table_setting = on;
+
+CREATE TABLE ao_alter_add_part(a int, b int) WITH (appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536) DISTRIBUTED BY (a) PARTITION BY RANGE (b)
+  (PARTITION "10" START (1) INCLUSIVE END (10) EXCLUSIVE,
+   PARTITION "20" START (11) INCLUSIVE END (20) EXCLUSIVE);
+SET gp_default_storage_options = 'appendonly=true, orientation=row, compresstype=zlib, compresslevel=2';
+-- use statement encoding
+ALTER TABLE ao_alter_add_part ADD PARTITION "30" START (20) INCLUSIVE END (30) EXCLUSIVE WITH (appendonly=true, orientation=row, compresstype=zlib, compresslevel=3, blocksize=16384);
+ALTER TABLE ao_alter_add_part ADD PARTITION "60" START (50) INCLUSIVE END (60) EXCLUSIVE WITH (appendonly=true, orientation=row);
+-- use table setting
+ALTER TABLE ao_alter_add_part ADD PARTITION "40" START (30) INCLUSIVE END (40) EXCLUSIVE;
+RESET gp_default_storage_options;
+-- use table setting
+ALTER TABLE ao_alter_add_part ADD PARTITION "50" START (40) INCLUSIVE END (50) EXCLUSIVE;
+\d+ ao_alter_add_part*
+DROP TABLE ao_alter_add_part;
+
+CREATE TABLE ao_alter_add_part_no_compress(a int, b int) WITH (appendonly=true, orientation=row) DISTRIBUTED BY (a) PARTITION BY RANGE (b)
+  (PARTITION "10" START (1) INCLUSIVE END (10) EXCLUSIVE,
+   PARTITION "20" START (11) INCLUSIVE END (20) EXCLUSIVE);
+SET gp_default_storage_options ='appendonly=true, orientation=row, compresstype=zlib, compresslevel=2, blocksize=8192';
+-- use statement encoding
+ALTER TABLE ao_alter_add_part_no_compress ADD PARTITION "30" START (20) INCLUSIVE END (30) EXCLUSIVE WITH (appendonly=true, orientation=row, compresstype=zstd, compresslevel=3, blocksize=16384);
+ALTER TABLE ao_alter_add_part_no_compress ADD PARTITION "60" START (50) INCLUSIVE END (60) EXCLUSIVE WITH (appendonly=true, orientation=row);
+-- use gp_default_storage_options
+ALTER TABLE ao_alter_add_part_no_compress ADD PARTITION "40" START (30) INCLUSIVE END (40) EXCLUSIVE;
+RESET gp_default_storage_options;
+-- use default value
+ALTER TABLE ao_alter_add_part_no_compress ADD PARTITION "50" START (40) INCLUSIVE END (50) EXCLUSIVE;
+\d+ ao_alter_add_part_no_compress*
+DROP TABLE ao_alter_add_part_no_compress;
+
+RESET gp_add_partition_inherits_table_setting;
```

Comment:
why do we need a test for the case there is no compression in the root table? is compression handled in some special way?

Reply:
> why do we need a test for the case there is no compression in the root table?

Yes, because without compression we may not point many options for it